test/foundry/deploy/19_ERC721H_tests.t.sol

```diff
@@ -0,0 +1,208 @@
+  MockExternalCall mockExternalCall;
+  ERC721H erc721h;
+  bytes initCode;
+  uint256 localHostFork;
+  string LOCALHOST_RPC_URL = vm.envString("LOCALHOST_RPC_URL");
+  address deployer = vm.addr(Constants.getPKDeployer());
+  address alice = vm.addr(1);
+
+  /**
+   * @notice Sets up the test environment
+   * @dev This function sets up the test environment by creating a local fork of the network,
+   * electing the fork, and initializing the ERC721H contract and the MockExternalCall contract.
+   */
+  function setUp() public {
+    localHostFork = vm.createFork(LOCALHOST_RPC_URL);
+    vm.selectFork(localHostFork);
+    erc721h = ERC721H(payable(Constants.getSampleERC721()));
+    mockExternalCall = new MockExternalCall();
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   init()                                   */
+  /* -------------------------------------------------------------------------- */
+  /**
+   * @notice Tests that the init function reverts when called twice
+   * @dev This test verifies that the init function of the ERC721 contract reverts when called twice.
+   * Refers to the hardhat test with the description 'should fail be initialized twice'
+   */
+  function testInit() public {
+    vm.expectRevert("HOLOGRAPHER: already initialized");
+    vm.prank(deployer);
+    erc721h.init(abi.encode(deployer));
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   owner()                                  */
+  /* -------------------------------------------------------------------------- */
+  /**
+   * @notice Tests that the owner function returns the correct owner
+   * @dev This test verifies that the owner function of the ERC721 contract returns the correct owner address.
+   * Refers to the hardhat test with the description 'should return the correct owner address'
+   */
+  function testReturnCorrectOwner() public {
+    vm.prank(deployer);
+    address ownerAddress = erc721h.owner();
+    assertEq(ownerAddress, deployer);
+  }
+
+  /**
+   * @notice Tests that the owner function does not return the wrong address
+   * @dev This test verifies that the owner function of the ERC721 contract does not return the wrong address.
+   * Refers to the hardhat test with the description 'should fail when comparing to wrong address'
+   */
+  function testComparingToWrongAddressFail() public {
+    vm.prank(alice);
+    address ownerAddress = erc721h.owner();
+    assertNotEq(ownerAddress, alice);
+  }
+
+  /**
+   * @notice Tests that an external contract can call the owner function
+   * @dev This test verifies that an external contract can successfully call the owner function of the ERC721 contract.
+   * Refers to the hardhat test with the description 'should allow external contract to call fn'
+   */
+  function testAllowExternalContractToCallFnOwner() public {
```

Comment:
I guess if unsuccessful it would revert, right? add this comment in the NatSpec

Reply:
Done